src/custom/Feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,231 @@
+
+const tooltipContent = (
+  <MeetWrapper>
+    Some account and system information may be sent to Layer5. We will use it to fix problems and
+    improve our services, subject to our{' '}
+    <StyledLink target="_blank" href="https://layer5.io/company/legal/privacy">
+      Privacy Policy
+    </StyledLink>{' '}
+    and{' '}
+    <StyledLink target="_blank" href="https://layer5.io/company/legal/terms-of-service">
+      Terms of Service
+    </StyledLink>
+    . We may email you for more information or updates.
+  </MeetWrapper>
+);
+
+interface FeedbackDataItem {
+  icon: JSX.Element;
+  label: string;
+  placeholder: string;
+  isTextInput: boolean;
+  innerComponent?: JSX.Element;
+}
+
+const feedbackData: FeedbackDataItem[] = [
+  {
+    icon: <FeedbackIcon />,
+    label: 'Issue',
+    placeholder: 'I’m having an issue with...',
+    isTextInput: true
+  },
+  {
+    icon: <IdeaIcon />,
+    label: 'Suggestion',
+    placeholder: 'I have a suggestion about...',
+    isTextInput: true
+  },
+  {
+    icon: <CalenderIcon />,
+    label: 'Meet Request',
+    placeholder: 'You have a suggestion3?',
+    isTextInput: false,
+    innerComponent: (
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+          height: '137px',
+          color: 'black'
+        }}
+      >
+        <Typography style={{ lineHeight: '2.5', textAlign: 'center' }}>
+          Need help or have more feedback than fits here?
+          <br /> Meet with us.
+        </Typography>
+        <StyledLink
+          target="_blank"
+          href="https://calendar.google.com/calendar/appointments/schedules/AcZssZ3pmcApaDP4xd8hvG5fy8ylxuFxD3akIRc5vpWJ60q-HemQi80SFFAVftbiIsq9pgiA2o8yvU56"
+        >
+          Select a time convenient for you.
+        </StyledLink>
+      </div>
+    )
+  }
+];
+
+interface FeedbackComponentProps {
+  onSubmit: (data: { label: string; message: string }) => void;
+}
+
+const FeedbackComponent: React.FC<FeedbackComponentProps> = ({ onSubmit }) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [submitted, setSubmitted] = useState<boolean>(false);
+  const [category, setCategory] = useState<FeedbackDataItem | undefined>();
+  const [messageValue, setMessageValue] = useState<string | undefined>();
+  const feedbackTextRef = useRef<HTMLTextAreaElement>(null);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsChecked(event.target.checked);
+  };
+
+  const handleFeedback = () => {
+    setIsOpen(!isOpen);
+    setCategory(feedbackData[0]);
+    setSubmitted(false);
+    setMessageValue('');
+  };
+
+  const handleSubmit = () => {
+    setSubmitted(true);
+    if (messageValue && isChecked) {
+      onSubmit({ label: category?.label || '', message: messageValue });
+    }
+    setTimeout(() => {
```

Comment:
Ouch .. timeouts  . @captain-Akshay  what is the state policy against timeouts here